### DISCUSSION TITLE
APP rodando em Background para enviar localização

### DISCUSSION THREAD

**Discussion by 30103010** [Flutterando/forum#107] [Discussões gerais]
Olá, se alguem puder me ajudar no seguinte problema:

Preciso criar um aplicativo que envie a localizacao do celular de 2 em 2 minutos para uma API, para isso usei um Timer.periodic , porem quando o app entra em segundo plano o Timer para de contar.  Depois disso eu coloquei o Timer.periodic dentro de uma Stream , isso funcionou parcialmente, porque com o cabo ligado no celular ele funciona , porem quando tiro o cabo, fora do modo debug , eu nao recebo as localizacoes. tentei liberar as permissões de bateria e localizacoes porem nao deu certo.
Depois dessas tentativas frustrantes, tentei o plugin Workmanager, ele funciona , porem consigo enviar as localizacoes somente de 15 em 15 minutos, pois é o maximo que o android permite após o android 8.  
Será que alguem consegue ter um Luz sobre esse assunto?  muito obrigado a todos.

**Comment by 30103010** [comment]
haaaa... detalhe , o que deu mais raiva é que um rapaz que trabalha comigo, montou um aplicativo simples em delphi gerando um log com as localizacoes, como exemplo, e em delphi nao ta dando perda, Ai fui obrigado a ouvir ( -EM DELPHI FUNCIONA !!! ) que raiva kkkkkkk.

**Comment by eduardoflorence** [comment]
Olá @30103010,

 O plugin que sempre ouço falar bem para esta funcionalidade é este: 
 https://pub.dev/packages/flutter_background_geolocation

**Comment by 30103010** [comment]
entao, essa ferramenta é paga para colocar em producao , e nao é barato nao kkkkkkk

**Comment by joelsongsouzza** [comment]
Acredito que agora com o Flutter 3.7, você consiga isso. Nessa atualização, os Isolates podem se comunicar com os Platform Channels.
Vê se [esse link ](https://medium.com/flutter/introducing-background-isolate-channels-7a299609cad8)pode te ajudar
